src/core/metrics/app-metrics.ts

```diff
@@ -170,16 +201,29 @@ export class AppMetrics {
       name,
       value: timing,
     };
+
     if (KONTUR_METRICS_DEBUG) {
       console.warn('metrics', payload);
     }
-    // TODO: use apiClientInstance after app init refactor
-    fetch(APP_METRICS_ENDPOINT, {
-      method: 'POST',
-      headers: { 'Content-Type': 'application/json' },
-      body: JSON.stringify([payload]),
-    }).catch((error) => {
-      console.error('sent metrics error:', error, payload);
+
+    this.reports.push(payload);
+  }
+
+  sendReports() {
+    apiClient.post(APP_METRICS_ENDPOINT, this.reports, true).catch((error) => {
+      console.error('error posting metrics :', error, this.reports);
     });
   }
 }
+
+function readyForScreenshot(map) {
+  // Still no reliable ways to detect when map is fully rendered, related issues are hanging for years
+  // TODO: implement better map rendered detection when possible
+  map.once('idle', function () {
+    setTimeout(() => {
+      const eventReadyEvent = new Event('event_ready_for_screenshot');
+      globalThis.dispatchEvent(eventReadyEvent);
+    }, 99); // extra time to prevent rendering glitches
+  });
+  map.triggerRepaint();
```

Comment:
so `triggerRepaint` will make sure we dispatch `eventReadyEvent`? Is that what we want?
Maybe another comment needed for that?

Reply:
Yes. And TODO is about that issue